### DISCUSSION TITLE
Added interface support for default implementation of auto attributes

### DISCUSSION THREAD

**Discussion by nczsl** [dotnet/csharplang#8023] [General]
It is recommended to support the implementation of the interface, and the contentless properties defined in the interface are not considered to be an interface implementation, but there is a need for this, which is valuable because it avoids duplicate implementation properties of the class of the interface

interface A{
  int X{get; set;} 
} // Now that this X is treated as a non-implementation, I can't find a way to make such a property be implemented by default, if it is the default implementation then it is much more convenient, right?

**Comment by CyrusNajmabadi** [comment]
@nczsl interfaces cannot have state.  There's no place for a field to go.  Adding state at the DIM level is something we also feel is highly problematic due to the diamond inheritance problem. 

Our recommendation here is to simply define an abstract class with an impl of all hte properties (which can be autoprops), and then derive from that.  As class inheritance doesn't allow for multiple-inheritance, there's no diamond problem here.

**Reply by nczsl** [reply]
Yes, my statement may not be very specific and accurate, in fact, my need is to let the class that inherits this interface not write these automatic properties repeatedly, not let the interface implement automatic properties

**Reply by CyrusNajmabadi** [reply]
@nczsl you could do this either with an abstract base class, or by writing a source generator to automatically emit the auto-props in the implementing class if you want to keep using interfaces.

**Reply by nczsl** [reply]
But there is a problem with the source code generator, 1 is to limit the type of library that must be net standard 2.0, and this kind of library has basically been abandoned in the net follow-up;2 is I mainly use vscode as a development tool, so it is not easy to use the source code generator

**Reply by CyrusNajmabadi** [reply]
> But there is a problem with the source code generator, 1 is to limit the type of library that must be net standard 2.0

This is only the code for the generator itself. The generator will still run even if you are compiling something much more recent.

**Reply by CyrusNajmabadi** [reply]
> is I mainly use vscode as a development tool, so it is not easy to use the source code generator

Generators work regardless of your ide host.

**Comment by nczsl** [comment]
So, if there is a field of the same type in the implementation class with the same name and lowercase, then if the class has the corresponding automatic property in the implemented interface, otherwise it is still handled according to the original rules, that is, the automatic property of the interface is still not implemented, unless there is a corresponding field of the same type with the same name (starting with lowercase) in the inheritance class.
